core/src/wallet/common/observers/AbstractBlockchainObserver.h

```diff
@@ -89,6 +89,18 @@ namespace ledger {
             virtual void putTransaction(const BlockchainExplorerTransaction& tx) = 0;
             virtual void putBlock(const BlockchainExplorerBlock& block) = 0;
 
+            static void emitEvent(const std::shared_ptr<Account> &account, std::function<bool (soci::session &sql, const std::shared_ptr<Account> &account)> eventEmission) {
+                bool shouldEmitNow = false;
+                {
+                    soci::session sql(account->getWallet()->getDatabase()->getPool());
+                    soci::transaction tr(sql);
+                    shouldEmitNow = eventEmission(sql, account);
+                    tr.commit();
+                }
```

Comment:
Maybe an explication is necessary here; why are you using nested scope ? Does `account->emitEventsNow()` have to wait the destruction of the transaction ?

Reply:
Yes we destruct the transaction before emitting to release lock on transactional level, because the emit might or not take time and it's unrelated to database so ...

Reply:
The emit can cause unexpected behaviour if the transaction isn't destroyed first ? If yes, I highly suggest to put a comment explaining why we're doing this that way and avoid future regression; it's easy to break the code by refactoring it (remove the nested scope for instance).

Reply:
It's not about the `emitEventsNow` having an unexpected behaviour on transactions, it's more that it could be blocking for destroying the transaction which you don't want, it's nothing specific to our use case but it's a more general rationale: if you have a resource to release you don't block it with a task which is completely independent, and refactos are not supposed just to remove scopes specially if it's meant to handle critical resources like `sql` transactions and connections.